server/config.js

```diff
@@ -12,22 +14,21 @@ export default function getConfig (dir) {
   return cache.get(dir)
 }
 
-async function loadConfig (dir) {
-  const path = join(dir, 'package.json')
+function loadConfig (dir) {
+  const path = join(dir, 'next.config.js')
 
-  let data
+  let module
   try {
-    data = await readFile(path, 'utf8')
+    module = require(path)
   } catch (err) {
-    if (err.code === 'ENOENT') {
-      data = '{}'
+    if (err.code === 'MODULE_NOT_FOUND') {
+      module = {}
     } else {
       throw err
     }
   }
 
-  // no try-cache, it must be a valid json
-  const config = JSON.parse(data).next || {}
+  const config = module.default || module || {}
```

Comment:
We we should also read the `package.json` and check for the next config. If it's there, we should get it. (Only the `cdn` field).
And if we found it, we need log a message as it's deprecated.